pkg/kubelet/status/status_manager.go

```diff
@@ -136,11 +141,17 @@ type Manager interface {
 	// the provided podUIDs.
 	RemoveOrphanedStatuses(podUIDs map[types.UID]bool)
 
-	// GetPodResizeStatus returns cached PodStatus.Resize value
-	GetPodResizeStatus(podUID types.UID) v1.PodResizeStatus
+	// GetPodResizeStatus returns cached PodStatus Resize conditions value
+	GetPodResizeStatus(podUID types.UID) PodResizeStatus
+
+	// SetPodResizePendingCondition caches the last PodResizePending condition for the pod.
+	SetPodResizePendingCondition(podUID types.UID, resizePending *v1.PodCondition)
 
-	// SetPodResizeStatus caches the last resizing decision for the pod.
-	SetPodResizeStatus(podUID types.UID, resize v1.PodResizeStatus)
+	// SetPodResizingCondition caches the last PodResizing condition for the pod.
+	SetPodResizingCondition(podUID types.UID, resizing *v1.PodCondition)
+
+	// ClearPodResizingConditions clears the PodResizingPending and PodResizing conditions for the pod.
+	ClearPodResizingConditions(podUID types.UID)
```

Comment:
We need to clear the conditions independently. You could use dedicated methods, or add a parameter to the Set*Condition methods, or have an implicit empty reason = clear it.

Reply:
sg, added dedicated method